internal/config/config_test.go

```diff
@@ -150,6 +151,50 @@ func TestLoadConfig(t *testing.T) {
 	}
 }
 
+func TestToIni(t *testing.T) {
+	expiration := 90
+	aad := config.AAD{TenantID: "tenantID", AppID: "appID", HomeDirPattern: "homeDirPattern", Shell: "shell", OfflineCredentialsExpiration: &expiration}
+
+	want := ini.Empty()
+	err := ini.ReflectFrom(want, &aad)
+	require.NoError(t, err, "Setup: failed to reflect config to ini")
+
+	got, err := aad.ToIni()
+	require.NoError(t, err, "Setup: failed to reflect config to ini")
+
+	require.Equal(t, want, got, "Got and expected ini files are different")
+}
+
+func TestValidate(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		configFile string
+		wantErr    bool
+	}{
+		"valid config, default domain":   {configFile: "valid.conf"},
+		"valid config, multiple domains": {configFile: "valid-multiple-domains.conf"},
```

Comment:
Thinking about it, do you check that there are required fields uncommented? I think Validate() should do this.

Reply:
Indeed, mind adding a test for it? (there is one when loading the config, but I guess it’s a great illustration of its usage)